File: src/Tests/Behavioral/MethodExprDotImport/mep/mep.go
// The FOREIGN package whose pointer-receiver methods the dot-importing main uses as method
// EXPRESSIONS. Each emits as a [GoRecv] extension static (plus a RecvGenerator ж-overload) in
// the mep_package C# class, so a method expression that binds them must QUALIFY the static form
// with this package's alias.
package mep

type Reader struct {
	Name string
}

func (r *Reader) Read(delim byte) (string, error) {
	return r.Name, nil
}

func (r *Reader) Peek(delim byte) (string, error) {
	return "peek:" + r.Name, nil
}

File: src/Tests/Behavioral/MethodExprDotImport/mep/mep.cs
namespace go.MethodExprDotImport;

partial class mep_package {

[GoType] partial struct Reader {
    public @string Name;
}

[GoRecv] public static (@string, error) Read(this ref Reader r, byte delim) {
    return (r.Name, default!);
}

[GoRecv] public static (@string, error) Peek(this ref Reader r, byte delim) {
    return ("peek:" + r.Name, default!);
}

} // end mep_package
